# Déjame escuchar la música

# Contenido <a id='back'></a>

* [Introducción](#intro)
* [Etapa 1. Descripción de los datos](#data_review)
    * [Conclusiones](#data_review_conclusions)
* [Etapa 2. Preprocesamiento de datos](#data_preprocessing)
    * [2.1 Estilo del encabezado](#header_style)
    * [2.2 Valores ausentes](#missing_values)
    * [2.3 Duplicados](#duplicates)
    * [2.4 Conclusiones](#data_preprocessing_conclusions)
* [Etapa 3. Prueba de hipótesis](#hypothesis)
    * [3.1 Hipótesis 1: actividad de los usuarios y las usuarias en las dos ciudades](#activity)
* [Conclusiones](#end)

## Introducción <a id='intro'></a>
Como analista de datos, tu trabajo consiste en analizar datos para extraer información valiosa y tomar decisiones basadas en ellos. Esto implica diferentes etapas, como la descripción general de los datos, el preprocesamiento y la prueba de hipótesis.

Siempre que investigamos, necesitamos formular hipótesis que después podamos probar. A veces aceptamos estas hipótesis; otras veces, las rechazamos. Para tomar las decisiones correctas, una empresa debe ser capaz de entender si está haciendo las suposiciones correctas.

En este proyecto, compararás las preferencias musicales de las ciudades de Springfield y Shelbyville. Estudiarás datos reales de transmisión de música online para probar la hipótesis a continuación y comparar el comportamiento de los usuarios y las usuarias de estas dos ciudades.

### Objetivo:
Prueba la hipótesis:
1. La actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad.


### Etapas
Los datos del comportamiento del usuario se almacenan en el archivo `/datasets/music_project_en.csv`. No hay ninguna información sobre la calidad de los datos, así que necesitarás examinarlos antes de probar la hipótesis.

Primero, evaluarás la calidad de los datos y verás si los problemas son significativos. Entonces, durante el preprocesamiento de datos, tomarás en cuenta los problemas más críticos.

Tu proyecto consistirá en tres etapas:
 1. Descripción de los datos.
 2. Preprocesamiento de datos.
 3. Prueba de hipótesis.








[Volver a Contenidos](#back)

## Etapa 1. Descripción de los datos <a id='data_review'></a>

Abre los datos y examínalos.

Necesitarás `pandas`, así que impórtalo.

In [10]:
# Importar pandas
import pandas as pd

Lee el archivo `music_project_en.csv` de la carpeta `/datasets/` y guárdalo en la variable `df`:

In [11]:
# Leer el archivo y almacenarlo en music_df
music_df = pd.read_csv('music_project_en(1).csv')
display(music_df)

userID                              Track            artist  \
0      FFB692EC                  Kamigata To Boots  The Mass Missile   
1      55204538        Delayed Because of Accident  Andreas Rönnberg   
2        20EC38                  Funiculì funiculà       Mario Lanza   
3      A3DD03C9              Dragons in the Sunset        Fire + Ice   
4      E2DC1FAE                        Soul People        Space Echo   
...         ...                                ...               ...   
65074  729CBB09                            My Name            McLean   
65075  D08D4A55  Maybe One Day (feat. Black Spade)       Blu & Exile   
65076  C5E3A0D5                          Jalopiina               NaN   
65077  321D0506                      Freight Train     Chas McDevitt   
65078  3A64EF84          Tell Me Sweet Little Lies      Monica Lopez   

            genre       City        time        Day  
0            rock  Shelbyville  20:28:33  Wednesday  
1            rock  Springfield  14:07:09     Friday  
2             pop  Shelbyville  20:58:07  Wednesday  
3            folk  Shelbyville  08:37:09     Monday  
4           dance  Springfield  08:34:34     Monday  
...           ...          ...       ...        ...  
65074         rnb  Springfield  13:32:28  Wednesday  
65075         hip  Shelbyville  10:00:00     Monday  
65076  industrial  Springfield  20:09:26     Friday  
65077        rock  Springfield  21:43:59     Friday  
65078     country  Springfield  21:59:46     Friday  

[65079 rows x 7 columns]

Muestra las 10 primeras filas de la tabla:

In [12]:
# Obtener las 10 primeras filas de la tabla music_df
display(music_df.head(10))

userID                        Track            artist   genre  \
0  FFB692EC            Kamigata To Boots  The Mass Missile    rock   
1  55204538  Delayed Because of Accident  Andreas Rönnberg    rock   
2    20EC38            Funiculì funiculà       Mario Lanza     pop   
3  A3DD03C9        Dragons in the Sunset        Fire + Ice    folk   
4  E2DC1FAE                  Soul People        Space Echo   dance   
5  842029A1                       Chains          Obladaet  rusrap   
6  4CB90AA5                         True      Roman Messer   dance   
7  F03E1C1F             Feeling This Way   Polina Griffith   dance   
8  8FA1D3BE                     L’estate       Julia Dalia  ruspop   
9  E772D5C0                    Pessimist               NaN   dance   

        City        time        Day  
0  Shelbyville  20:28:33  Wednesday  
1  Springfield  14:07:09     Friday  
2  Shelbyville  20:58:07  Wednesday  
3  Shelbyville  08:37:09     Monday  
4  Springfield  08:34:34     Monday  
5  Shelbyville  13:09:41     Friday  
6  Springfield  13:00:07  Wednesday  
7  Springfield  20:47:49  Wednesday  
8  Springfield  09:17:40     Friday  
9  Shelbyville  21:20:49  Wednesday

Obtén la información general sobre la tabla con un comando. Conoces el método que muestra la información general que necesitamos.

In [13]:
# Obtener la información general sobre nuestros datos
music_df.info()
print(music_df.isna().mean())
print()
print('cantidad de duplicados:')
print(music_df.duplicated().sum())

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 65079 entries, 0 to 65078
Data columns (total 7 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0     userID  65079 non-null  object
 1   Track     63736 non-null  object
 2   artist    57512 non-null  object
 3   genre     63881 non-null  object
 4     City    65079 non-null  object
 5   time      65079 non-null  object
 6   Day       65079 non-null  object
dtypes: object(7)
memory usage: 3.5+ MB
  userID    0.000000
Track       0.020636
artist      0.116274
genre       0.018408
  City      0.000000
time        0.000000
Day         0.000000
dtype: float64

cantidad de duplicados:
3826


Estas son nuestras observaciones sobre la tabla. Contiene siete columnas. Almacenan los mismos tipos de datos: `object`.

Según la documentación:
- `' userID'`: identificador del usuario o la usuaria;
- `'Track'`: título de la canción;
- `'artist'`: nombre del artista;
- `'genre'`: género de la pista;
- `'City'`: ciudad del usuario o la usuaria;
- `'time'`: la hora exacta en la que se reprodujo la canción;
- `'Day'`: día de la semana.

Podemos ver tres problemas con el estilo en los encabezados de la tabla:
1. Algunos encabezados están en mayúsculas, otros en minúsculas.
2. Hay espacios en algunos encabezados.
3. `Detecta el tercer problema por tu cuenta y descríbelo aquí`.


EL TERCER PROBLEMA: 
hay nulos e inconsistencia en el tipo de dato de la columna tiempo


### Escribe observaciones de tu parte. Estas son algunas de las preguntas que pueden ser útiles: <a id='data_review_conclusions'></a>

`1.   ¿Qué tipo de datos tenemos a nuestra disposición en las filas? ¿Y cómo podemos entender lo que almacenan las columnas?`

En las filas, tenemos datos deregistros de reproducción de canciones con información sobre el usuario, la canción, el artista, el género musical, la ciudad del usuario y la hora y día de la semana en que se reprodujo la canción.(' userID'
'Track', 'artist', 'genre', 'City', 'time' y 'Day')  
`2.   ¿Hay suficientes datos para proporcionar respuestas a nuestra hipótesis o necesitamos más información?`
Con más de 65,000 filas, parece haber suficientes datos para analizar tendencias y patrones en la reproducción de canciones. Sin embargo, la suficiencia de los datos dependerá de la hipótesis específica que se quiera probar. Es posible que se necesiten más datos para analizar aspectos más detallados o específicos, pero talves halla los suficientes pero una explicacion de otra forma los dejaria mas en claro para cualquier usuario aveces algunos usuaruios con los los mismos datos pero con diferente explicacion lo entiendan mejor


`3.   ¿Notaste algún problema en los datos, como valores ausentes, duplicados o tipos de datos incorrectos?`
  si HAY VALORES AUSENTES Track artist genre DUPLICADOS 3826
  el tiempo esta incorrecto pero el analisis no requiere el cambio

[Volver a Contenidos](#back)

## Etapa 2. Preprocesamiento de datos <a id='data_preprocessing'></a>

El objetivo aquí es preparar los datos para que sean analizados.
El primer paso es resolver cualquier problema con los encabezados. Luego podemos avanzar a los valores ausentes y duplicados. Empecemos.

Corrige el formato en los encabezados de la tabla.


### Estilo del encabezado <a id='header_style'></a>
Muestra los encabezados de la tabla (los nombres de las columnas):

In [14]:


# Mostrar los nombres de las columnas
print(music_df.columns)

Index(['  userID', 'Track', 'artist', 'genre', '  City  ', 'time', 'Day'], dtype='object')


Cambia los encabezados de la tabla de acuerdo con las reglas del buen estilo:
* Todos los caracteres deben ser minúsculas.
* Elimina los espacios.
* Si el nombre tiene varias palabras, utiliza snake_case.

Anteriormente, aprendiste acerca de la forma automática de cambiar el nombre de las columnas. Vamos a aplicarla ahora. Utiliza el bucle for para iterar sobre los nombres de las columnas y poner todos los caracteres en minúsculas. Cuando hayas terminado, vuelve a mostrar los encabezados de la tabla:

In [15]:
# Bucle en los encabezados poniendo todo en minúsculas
for i, col in enumerate(music_df.columns):
    music_df.rename(columns={col: col.strip().lower()}, inplace=True)

Ahora, utilizando el mismo método, elimina los espacios al principio y al final de los nombres de las columnas e imprime los nombres de las columnas nuevamente:

In [16]:
# Bucle en los encabezados eliminando los espacios
music_df.columns

Index(['userid', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')

Necesitamos aplicar la regla de snake_case a la columna `userid`. Debe ser `user_id`. Cambia el nombre de esta columna y muestra los nombres de todas las columnas cuando hayas terminado.

In [17]:
# Cambiar el nombre de la columna "userid"
music_df = music_df.rename(columns={'userid': 'user_id'})

Comprueba el resultado. Muestra los encabezados una vez más:

In [18]:
# Comprobar el resultado: la lista de encabezados
print(music_df.columns)

Index(['user_id', 'track', 'artist', 'genre', 'city', 'time', 'day'], dtype='object')


[Volver a Contenidos](#back)

### Valores ausentes <a id='missing_values'></a>
 Primero, encuentra el número de valores ausentes en la tabla. Debes utilizar dos métodos en una secuencia para obtener el número de valores ausentes.

In [19]:
# Calcular el número de valores ausentes
print(music_df.isnull().sum())

user_id       0
track      1343
artist     7567
genre      1198
city          0
time          0
day           0
dtype: int64


No todos los valores ausentes afectan a la investigación. Por ejemplo, los valores ausentes en `track` y `artist` no son cruciales. Simplemente puedes reemplazarlos con valores predeterminados como el string `'unknown'` (desconocido).

Pero los valores ausentes en `'genre'` pueden afectar la comparación entre las preferencias musicales de Springfield y Shelbyville. En la vida real, sería útil saber las razones por las cuales hay datos ausentes e intentar recuperarlos. Pero no tenemos esa oportunidad en este proyecto. Así que tendrás que:
* rellenar estos valores ausentes con un valor predeterminado;
* evaluar cuánto podrían afectar los valores ausentes a tus cómputos;

Reemplazar los valores ausentes en las columnas `'track'`, `'artist'` y `'genre'` con el string `'unknown'`. Como mostramos anteriormente en las lecciones, la mejor forma de hacerlo es crear una lista que almacene los nombres de las columnas donde se necesita el reemplazo. Luego, utiliza esta lista e itera sobre las columnas donde se necesita el reemplazo haciendo el propio reemplazo.

In [20]:
# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'
# Crear una lista con los nombres de las columnas donde se necesita el reemplazo
columns_to_replace = ['track', 'artist', 'genre']

# Bucle en los encabezados reemplazando los valores ausentes con 'unknown'
for column in columns_to_replace:
    music_df[column] = music_df[column].fillna('unknown')

Ahora comprueba el resultado para asegurarte de que después del reemplazo no haya valores ausentes en el conjunto de datos. Para hacer esto, cuenta los valores ausentes nuevamente.

In [21]:
# Contar valores ausentes
print(music_df.isnull().sum())

user_id    0
track      0
artist     0
genre      0
city       0
time       0
day        0
dtype: int64


[Volver a Contenidos](#back)

### Duplicados <a id='duplicates'></a>
Encuentra el número de duplicados explícitos en la tabla. Una vez más, debes aplicar dos métodos en una secuencia para obtener la cantidad de duplicados explícitos.

In [22]:
# Contar duplicados explícitos
print(music_df.duplicated().sum())

3826


Ahora, elimina todos los duplicados. Para ello, llama al método que hace exactamente esto.

In [23]:
# Eliminar duplicados explicitos
print(music_df.drop_duplicates(inplace=True))


None


Comprobemos ahora si eliminamos con éxito todos los duplicados. Cuenta los duplicados explícitos una vez más para asegurarte de haberlos eliminado todos:

In [24]:
# Comprobar de nuevo si hay duplicados

print(music_df.duplicated().sum())

0


Ahora queremos deshacernos de los duplicados implícitos en la columna `genre`. Por ejemplo, el nombre de un género se puede escribir de varias formas. Dichos errores también pueden afectar al resultado.

Para hacerlo, primero mostremos una lista de nombres de género únicos, ordenados en orden alfabético. Para ello:
* Extrae la columna `genre` del DataFrame.
* Llama al método que devolverá todos los valores únicos en la columna extraída.


In [25]:
# Inspeccionar los nombres de géneros únicos
print(sorted(music_df['genre'].unique()))

['acid', 'acoustic', 'action', 'adult', 'africa', 'afrikaans', 'alternative', 'ambient', 'americana', 'animated', 'anime', 'arabesk', 'arabic', 'arena', 'argentinetango', 'art', 'audiobook', 'avantgarde', 'axé', 'baile', 'balkan', 'beats', 'bigroom', 'black', 'bluegrass', 'blues', 'bollywood', 'bossa', 'brazilian', 'breakbeat', 'breaks', 'broadway', 'cantautori', 'cantopop', 'canzone', 'caribbean', 'caucasian', 'celtic', 'chamber', 'children', 'chill', 'chinese', 'choral', 'christian', 'christmas', 'classical', 'classicmetal', 'club', 'colombian', 'comedy', 'conjazz', 'contemporary', 'country', 'cuban', 'dance', 'dancehall', 'dancepop', 'dark', 'death', 'deep', 'deutschrock', 'deutschspr', 'dirty', 'disco', 'dnb', 'documentary', 'downbeat', 'downtempo', 'drum', 'dub', 'dubstep', 'eastern', 'easy', 'electronic', 'electropop', 'emo', 'entehno', 'epicmetal', 'estrada', 'ethnic', 'eurofolk', 'european', 'experimental', 'extrememetal', 'fado', 'film', 'fitness', 'flamenco', 'folk', 'folklor

Busca en la lista para encontrar duplicados implícitos del género `hiphop`. Estos pueden ser nombres escritos incorrectamente o nombres alternativos para el mismo género.

Verás los siguientes duplicados implícitos:
* `hip`
* `hop`
* `hip-hop`

Para deshacerte de ellos, crea una función llamada `replace_wrong_genres()` con dos parámetros:
* `wrong_genres=`: esta es una lista que contiene todos los valores que necesitas reemplazar.
* `correct_genre=`: este es un string que vas a utilizar como reemplazo.

Como resultado, la función debería corregir los nombres en la columna `'genre'` de la tabla `df`, es decir, remplazar cada valor de la lista `wrong_genres` por el valor en `correct_genre`.

Dentro del cuerpo de la función, utiliza un bucle `'for'` para iterar sobre la lista de géneros incorrectos, extrae la columna `'genre'` y aplica el método `replace` para hacer correcciones.

In [26]:
# Función para reemplazar duplicados implícitos
def replace_wrong_genres(wrong_genres,correct_genre):
    music_df['genre']= music_df['genre'].replace(wrong_genres,correct_genre)
    return music_df

Ahora, llama a `replace_wrong_genres()` y pásale tales argumentos para que retire los duplicados implícitos (`hip`, `hop` y `hip-hop`) y los reemplace por `hiphop`:

In [27]:
# Eliminar duplicados implícitos
music_df=replace_wrong_genres(['hip','hop','hip-hop'],'hiphop')

Asegúrate de que los nombres duplicados han sido eliminados. Muestra la lista de valores únicos de la columna `'genre'` una vez más:

In [28]:
# Comprobación de duplicados implícitos
sorted(music_df['genre'].unique())

['acid',
 'acoustic',
 'action',
 'adult',
 'africa',
 'afrikaans',
 'alternative',
 'ambient',
 'americana',
 'animated',
 'anime',
 'arabesk',
 'arabic',
 'arena',
 'argentinetango',
 'art',
 'audiobook',
 'avantgarde',
 'axé',
 'baile',
 'balkan',
 'beats',
 'bigroom',
 'black',
 'bluegrass',
 'blues',
 'bollywood',
 'bossa',
 'brazilian',
 'breakbeat',
 'breaks',
 'broadway',
 'cantautori',
 'cantopop',
 'canzone',
 'caribbean',
 'caucasian',
 'celtic',
 'chamber',
 'children',
 'chill',
 'chinese',
 'choral',
 'christian',
 'christmas',
 'classical',
 'classicmetal',
 'club',
 'colombian',
 'comedy',
 'conjazz',
 'contemporary',
 'country',
 'cuban',
 'dance',
 'dancehall',
 'dancepop',
 'dark',
 'death',
 'deep',
 'deutschrock',
 'deutschspr',
 'dirty',
 'disco',
 'dnb',
 'documentary',
 'downbeat',
 'downtempo',
 'drum',
 'dub',
 'dubstep',
 'eastern',
 'easy',
 'electronic',
 'electropop',
 'emo',
 'entehno',
 'epicmetal',
 'estrada',
 'ethnic',
 'eurofolk',
 'european',
 'expe

[Volver a Contenidos](#back)

### Tus observaciones <a id='data_preprocessing_conclusions'></a>

`Describe brevemente lo que has notado al analizar duplicados, cómo abordaste sus eliminaciones y qué resultados obtuviste.`
tilicé el método duplicated() para identificar los duplicados explícitos en la tabla.
Luego, utilicé el método drop_duplicates() para eliminar los duplicados explícitos.
Finalmente, verifiqué que los duplicados explícitos habían sido eliminados correctamente, utilizando nuevamente el método duplicated().
en los Duplicados Implícitos    
Identifiqué los duplicados implícitos en la columna 'genre', específicamente en los géneros 'hip', 'hop' y 'hip-hop'.
Creé una función llamada replace_wrong_genres() que reemplazó estos valores por 'hiphop'.
    Luego, llamé a la función replace_wrong_genres() y pasé los argumentos necesarios para reemplazar los duplicados implícitos.
    Finalmente, verifiqué que los duplicados implícitos habían sido eliminados correctamente, utilizando el método unique() para mostrar la lista de valores únicos en la columna 'genre'.




[Volver a Contenidos](#back)

## Etapa 3. Prueba de hipótesis <a id='hypothesis'></a>

### Hipótesis: comparar el comportamiento del usuario o la usuaria en las dos ciudades <a id='activity'></a>

La hipótesis afirma que existen diferencias en la forma en que los usuarios y las usuarias de Springfield y Shelbyville consumen música. Para comprobar esto, usa los datos de tres días de la semana: lunes, miércoles y viernes.

* Agrupa a los usuarios y las usuarias por ciudad.
* Compara el número de canciones que cada grupo reprodujo el lunes, el miércoles y el viernes.


Realiza cada cálculo por separado.

El primer paso es evaluar la actividad del usuario en cada ciudad. Recuerda las etapas dividir-aplicar-combinar de las que hablamos anteriormente en la lección. Tu objetivo ahora es agrupar los datos por ciudad, aplicar el método apropiado para contar durante la etapa de aplicación y luego encontrar la cantidad de canciones reproducidas en cada grupo especificando la columna para obtener el recuento.

A continuación se muestra un ejemplo de cómo debería verse el resultado final:
`df.groupby(by='....')['column'].method()`Realiza cada cálculo por separado.

Para evaluar la actividad de los usuarios y las usuarias en cada ciudad, agrupa los datos por ciudad y encuentra la cantidad de canciones reproducidas en cada grupo.



In [29]:
# Contar las canciones reproducidas en cada ciudad
music_df.groupby(by='city')['track'].count()

city
Shelbyville    18512
Springfield    42741
Name: track, dtype: int64

`Comenta tus observaciones aquí`
springfield parece tener mas reproduccines de tracks 

Ahora agrupemos los datos por día de la semana y encontremos el número de canciones reproducidas el lunes, miércoles y viernes. Utiliza el mismo método que antes, pero ahora necesitamos una agrupación diferente.


In [30]:
# Calcular las canciones reproducidas en cada uno de los tres días
music_df.groupby(by='day')['track'].count()

day
Friday       21840
Monday       21354
Wednesday    18059
Name: track, dtype: int64

`Comenta tus observaciones aquí` en viernes se reproducen mas canciones y en lunes menos eso quiere decir que entre mas se acerca el fin de semana mas hay la necesidad de escuchar musica

Ya sabes cómo contar entradas agrupándolas por ciudad o día. Ahora necesitas escribir una función que pueda contar entradas según ambos criterios simultáneamente.

Crea la función `number_tracks()` para calcular el número de canciones reproducidas en un determinado día **y** ciudad. La función debe aceptar dos parámetros:

- `day`: un día de la semana para filtrar. Por ejemplo, `'Monday'` (lunes).
- `city`: una ciudad para filtrar. Por ejemplo, `'Springfield'`.

Dentro de la función, aplicarás un filtrado consecutivo con indexación lógica.

Primero filtra los datos por día y luego filtra la tabla resultante por ciudad.

Después de filtrar los datos por dos criterios, cuenta el número de valores de la columna 'user_id' en la tabla resultante. Este recuento representa el número de entradas que estás buscando. Guarda el resultado en una nueva variable y devuélvelo desde la función.

In [31]:
# Declara la función number_tracks() con dos parámetros: day= y city=.
def number_tracks(day, city):
    # Almacena las filas del DataFrame donde el valor en la columna 'day' es igual al parámetro day=
    day=music_df_filtered_by_day = music_df[music_df['day'] == day]
    # Filtra las filas donde el valor en la columna 'city' es igual al parámetro city=
    music_df_filtered_by_day_and_city = music_df_filtered_by_day[music_df_filtered_by_day['city'] == city]
    # Extrae la columna 'user_id' de la tabla filtrada y aplica el método count()
    number_of_tracks = music_df_filtered_by_day_and_city['user_id'].count()
    # Devolve el número de valores de la columna 'user_id'
    return number_of_tracks

Llama a `number_tracks()` seis veces, cambiando los valores de los parámetros para que recuperes los datos de ambas ciudades para cada uno de los tres días.

In [32]:
# El número de canciones reproducidas en Springfield el lunes
springfield_monday = number_tracks('Monday', 'Springfield')
print(springfield_monday)

15740


In [33]:
# El número de canciones reproducidas en Shelbyville el lunes
shelbyville_monday = number_tracks('Monday', 'Shelbyville')
print(shelbyville_monday)

5614


In [34]:
# El número de canciones reproducidas en Springfield el miércoles
springfield_wednesday = number_tracks('Wednesday', 'Springfield')
print(springfield_wednesday)

11056


In [35]:
# El número de canciones reproducidas en Shelbyville el miércoles
shelbyville_wednesday = number_tracks('Wednesday', 'Shelbyville')
print(shelbyville_wednesday)

7003


In [36]:
# El número de canciones reproducidas en Springfield el viernes
springfield_friday = number_tracks('Friday', 'Springfield')
print(springfield_friday)

15945


In [37]:
# El número de canciones reproducidas en Shelbyville el viernes
shelbyville_friday = number_tracks('Friday', 'Shelbyville')
print(shelbyville_friday)

5895


**Conclusiones**

`Comenta si la hipótesis es correcta o se debe rechazar. Explica tu razonamiento.`
La hipótesis es correcta.


segun los numeros en los datos muestran que la actividad de los usuarios y las usuarias difiere según el día de la semana y dependiendo de la ciudad. 
En Springfield, el número de canciones reproducidas es mayor el viernes (15945) y menor el miércoles (11056). En Shelbyville, el número de canciones reproducidas también es mayor el viernes (5895) y menor el miércoles (7003). 
Además, en ambos días, Springfield tiene un número significativamente mayor de canciones reproducidas que Shelbyville. Esto sugiere que la actividad musical es más intensa en Springfield que en Shelbyville. una variacion notable

[Volver a Contenidos](#back)

# Conclusiones <a id='end'></a>

`Resume aquí tus conclusiones sobre la hipótesis.` 
el número de canciones reproducidas en ambas ciudades muestran su variacion notable y que ciudad es mejor para ofrecer musica en cualquier tipo de ventas y los dias ideales para ello. asi mismo cuando la puclicidad se es propicio poner en lo relacionado a la musica. DETALLE IMPORTANTE PARA REVISOR: CREOQUE ALTERE ALGUNA CELDA Y SE MODIFICO


### Nota
En proyectos de investigación reales, la prueba de hipótesis estadística es más precisa y cuantitativa. También ten en cuenta que no siempre se pueden sacar conclusiones sobre una ciudad entera a partir de datos de una sola fuente.

Aprenderás más sobre la prueba de hipótesis en el sprint de análisis estadístico de datos.

[Volver a Contenidos](#back)